pandas/core/style.py

```diff
@@ -0,0 +1,690 @@
+        else:
+            # like tee
+            result = func(self.data.loc[subset], **kwargs)
+        self._update_ctx(result)
+        return self
+
+    def apply(self, func, axis=0, subset=None, **kwargs):
+        """
+        Apply a function, updating the HTML representation with the result.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        func: function
+        axis: int, str or None
+            apply to each column (``axis=0`` or ``'index'``)
+            or to each row (``axis=1`` or ``'columns'``) or
+            to the entire DataFrame at once with ``axis=None``.
+        subset: IndexSlice
+            a valid indexer to limit ``data`` to *before* applying the
+            function. Consider using a pandas.IndexSlice
+        kwargs: dict
+            pass along to ``func``
+
+        Returns
+        -------
+        self
+
+        Notes
+        -----
+        This is similar to DataFrame.apply, except that axis=None applies
+        the function to the entire DataFrame at once, rather tha column
+        or rowwise.
+
+        Examples
+        --------
+
+        """
+        self._todo.append((lambda instance: getattr(instance, '_apply'),
+                          (func, axis, subset),
+                           kwargs))
+        return self
+
+    def _applymap(self, func, subset=None, **kwargs):
+        func = partial(func, **kwargs)  # applymap doesn't take kwargs?
+        if subset is None:
+            subset = pd.IndexSlice[:]
+        subset = _non_reducing_slice(subset)
+        result = self.data.loc[subset].applymap(func)
+        self._update_ctx(result)
+        return self
+
+    def applymap(self, func, subset=None, **kwargs):
+        """
+        Apply a function elementwise, updating the HTML
+        representation with the result.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        func : function
+        subset : IndexSlice
+            a valid indexer to limit ``data`` to *before* applying the
+            function. Consider using a pandas.IndexSlice
+        kwargs : dict
+            pass along to ``func``
+
+        Returns
+        -------
+        self
+
+        Notes
+        -----
+        Examples
+        --------
+        """
+        self._todo.append((lambda instance: getattr(instance, '_applymap'),
+                          (func, subset),
+                          kwargs))
+        return self
+
+    def set_precision(self, precision):
+        self.precision = precision
+        return self
+
+    def export(self):
+        return self._todo
+
+    def set(self, styles):
+        self._todo.extend(styles)
+        return self
+
+    def set_uuid(self, uuid):
+        self.uuid = uuid
+        return self
+
+    def set_caption(self, caption):
+        self.caption = caption
+        return self
+
+    def set_table_styles(self, table_styles):
```

Comment:
I think those two cover everything (unless I'm missing something).

- `.set_properties` takes `attribute: value` and sets it on the class for each cell.
- `.set_table_styles` takes (a list of) `selector: [(attribute, value)]` and puts that in a `<style>` tag before the table.

IIUC `.set_css` is stuff that would apply to the table? That's covered by `.set_table_styles{'': [(attrs, values)])}` i.e. no selector so you just have the id of the table.